React/Fabric/Mounting/ComponentViews/Modal/RCTModalHostViewComponentView.mm

```diff
@@ -192,9 +192,14 @@ - (void)ensurePresentedOnlyIfNeeded
 
 #pragma mark - RCTMountingTransactionObserving
 
-- (void)mountingTransactionWillMountWithMetadata:(MountingTransactionMetadata const &)metadata
+- (void)mountingTransactionWillMount:(MountingTransaction const &)transaction
 {
-  _modalContentsSnapshot = [self.viewController.view snapshotViewAfterScreenUpdates:NO];
+  for (auto mutation : transaction.getMutations()) {
```

Comment:
Is this additional logic required for the snapshot to work?
How did it work before assuming metadata was passed from the same place?

Reply:
It wasn't needed. We were just making those snapshots more often than needed. Apparently the snapshotting method can be quite costly so this actually optimizes the number of times we do the snapshot.

Before, we were making a snapshot for every single transaction. Including transactions that weren't even touching the modal component while we only needed the snapshot to be taken right before the component is removed. Adding this logic makes it so that we only snapshot when we know that transaction contains a remove mutation. This isn't still ideal as if there are more then one modals, we will actually snapshot all of them when one of the modals is removed (because the logic here only looks at view type of a given remove mutation), but this already is an improvement over previous behavior.

Reply:
Let's extract those changes into different PR maybe?
Just want to make sure we merge these changes incrementally :)